BACKEND/userAuth/src/main/java/com/apptime/auth/service/TaskManagerService.java

```diff
@@ -68,9 +70,48 @@ public Task deleteTask(long id) {
 		return taskRepo.findByUserName(user);
 	}
 
+	/**
+	 *
+	 * @param taskId
+	 * @param startDate
+	 * @return
+	 */
+	public TaskState start(long  taskId, Date startDate){
+		Task task = taskRepo.findById(taskId);
+		if(task != null){
+			TaskStateMachine.START(task);
+			task.setStart(startDate);
+			taskRepo.save(task);
+		}
+		return task.getState();
```

Comment:
If task is null, this line would throw `NullPointerException`.

Reply:
good catch. if it can't find the task now it returns null instead of throwing NullPointer